src/features/berth/selectedBerthPage/types.ts

```diff
@@ -0,0 +1,26 @@
+import { BerthSwitchReasonsQuery_berthSwitchReasons } from '../../__generated__/BerthSwitchReasonsQuery';
+
+type Option = {
+  label: string;
+  value: string;
+};
+
+export type BerthOption = Option;
+
+export type PierOption = Option & {
+  berths: BerthOption[];
+};
+
+export type HarborOption = Option & {
+  piers: PierOption[];
+};
+
+export type ReasonOption = Option;
+
+export type BoatInfo = {
+  boatType?: string | null;
```

Comment:
This won't let you pass an object without `boatType` property
```suggestion
  boatType: string | null | undefined;
```